src/mlpack/methods/preprocess/preprocess_string_main.cpp

```diff
@@ -0,0 +1,188 @@
+    "\n\n" +
+    PRINT_CALL("preprocess_string", "actual_dataset", "X",
+        "preprocess_dataset", "X", "lowercase", 1, "stopwords",
+        1, "punctuation", 1, "dimension", 3) +
+    "\n\n",
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+PARAM_STRING_IN_REQ("actual_dataset", "File containing the reference dataset.",
+    "t");
+PARAM_STRING_IN_REQ("preprocess_dataset", "File containing the preprocess "
+    "dataset.", "o");
+PARAM_STRING_IN_REQ("delimeter", "delimeter used to seperate Column in files"
+    "example '\\t' for '.tsv' and ',' for '.csv'.", "d");
+PARAM_STRING_IN("strDelimeter", "A set of chars that is used as delimeter to"
+    "tokenize the string dataset ", "D", " ");
+PARAM_STRING_IN("stopwordsfile", "File containing stopwords", "S", "");
+
+PARAM_FLAG("lowercase", "convert to lowercase.", "l");
+PARAM_FLAG("punctuation", "Remove punctuation.", "p");
+PARAM_FLAG("stopwords", "Remove stopwords.", "s");
+PARAM_VECTOR_IN_REQ(int, "dimension", "Column which contains the string data."
+    "(1 by default)", "c");
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+static void mlpackMain()
+{
+  // Parse command line options.
+  // Extracting the filename
+  const std::string filename = CLI::GetParam<std::string>("actual_dataset");
+  // This is very dangerous, Let's add a check tommorrow.
+  const std::string delimiter = CLI::GetParam<std::string>("delimeter");
+  std::vector<int> dimension = CLI::GetParam<std::vector<int>>("dimension");
+  // Sorting neccessary
+  std::sort(dimension.begin(), dimension.end());
+  // Extracting the Contents of file
+  // File pointer
+  fstream fin;
+  // Open an existing file
+  fin.open(filename, ios::in);
+  std::string line;
+  std::vector<std::string> temp;
+  std::vector<std::vector<std::string>> dataset;
+  // const size_t Column = CLI::GetParam<int>("dimension");
+  data::SplitByChar sobj(delimiter);
+  boost::string_view token, copy;
+  while (std::getline(fin, line))
+  {
+    copy = line;
+    token = sobj(copy);
+    while (!token.empty())
+    {
+      temp.push_back(std::string(token));
+      token = sobj(copy);
+    }
+    dataset.push_back(temp);
+    temp.clear();
+  }
+  fin.close();
+  // Preparing the input dataset on which string manipulation has to be done.
+  size_t row = 0;
+  std::vector<std::vector<std::string>> input(dimension.size());
+  for (size_t i = 0; i < dataset.size(); i++)
+  {
+    row = 0;
+    for (size_t j = 0; j < dataset[i].size(); j++)
+    {
+      if(row < dimension.size())
+      {
+        if (dimension[row] == j)
+        {
+          // insert into input
+          input[row].push_back(dataset[i][j]);
+          row++;          
+        }
+      }
+      else
+      {
+        break;
+      }
+    }
+  }
```

Comment:
I am not sure that I understand this loop. I thought you wanted to process some peculiar columns. Why do you verify row numbers?